src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -479,17 +465,17 @@ private void CreateFromBinaryForm(byte[] binaryForm, int offset)
                 (((long)binaryForm[offset + 6]) << 8) +
                 (((long)binaryForm[offset + 7])));
 
-            SubAuthorities = new int[binaryForm[offset + 1]];
+            int subAuthoritiesLength = binaryForm[offset + 1];
 
             //
             // Subauthorities are represented in big-endian format
             //
 
-            for (byte i = 0; i < binaryForm[offset + 1]; i++)
+            for (int i = 0; i < binaryForm[offset + 1]; i++)
```

Comment:
You factored the `binaryForm[offset + 1]` expression out to a local, but didn't use it here... is there a reason?

Reply:
Lack of focus. Fixed.